pkg/k8s/api.go

```diff
@@ -197,7 +198,7 @@ func (kubeAPI *KubernetesAPI) GetNamespaceWithExtensionLabel(ctx context.Context
 			return &ns, err
 		}
 	}
-	return nil, fmt.Errorf("could not find the %s extension", value)
+	return nil, errors.NewNotFound(corev1.Resource("namespace"), value)
```

Comment:
Now that the `linkerd check` check is removed we don't really need this change, but I'm okay still keeping it. If not now, I still think something in the future can use this error type check for differentiating between `NotFound` and an actual error.

Reply:
Yeah, Totally makes sense!